### ISSUE TITLE
Validation error: u'resource_type' is a required property

### ISSUE THREAD

**Issue by camarasu** [boutiques/boutiques#414]

**Comment by camarasu** [comment]
Issue encountered with boutiques 0.5.18: the descriptor is uploaded, but not published

bosh publish -v --sandbox /tmp/gate-opendose-descriptor.json
[ INFO ] Zenodo access token saved in /home/abonnet/.boutiques
[ INFO ] Using Zenodo endpoint https://sandbox.zenodo.org
The descriptor will be published to Zenodo, this cannot be undone. Are you sure? (Y/n) Y
[ INFO (401) ] Zenodo is accessible
[ INFO (200) ] Authentication to Zenodo successful
[ INFO ] Using Zenodo endpoint https://sandbox.zenodo.org
[ INFO (200) ] Search successful.
[ INFO (201) ] Deposition succeeded, id is 261389
[ INFO (201) ] Descriptor uploaded to Zenodo
[ INFO ] {u'status': 400, u'message': u"Validation error: u'resource_type' is a required property."}
[ ERROR (400) ] Cannot publish descriptor 

Note that the same descriptor is correctly published with boutiques 0.5.13
An example is https://zenodo.org/record/2573071

**Comment by glatard** [comment]
Hi @camarasu, I can't reproduce the issue:

```
$ bosh publish --sandbox -v /tmp/a.json 
[ INFO ] Zenodo access token saved in /home/glatard/.boutiques
[ INFO ] Using Zenodo endpoint https://sandbox.zenodo.org
The descriptor will be published to Zenodo, this cannot be undone. Are you sure? (Y/n) Y
[ INFO (401) ] Zenodo is accessible
[ INFO (200) ] Authentication to Zenodo successful
[ INFO ] Using Zenodo endpoint https://sandbox.zenodo.org
[ INFO (200) ] Search successful.
Found an existing record with the same name, would you like to update it? (Y:Update existing / n:Publish new entry with name GateCLforOpenDose) n
[ INFO (201) ] Deposition succeeded, id is 261431
[ INFO (201) ] Descriptor uploaded to Zenodo
[ INFO (202) ] Descriptor published to Zenodo, doi is 10.5072/zenodo.261431
10.5072/zenodo.261431
```
I tried with Python 2.7 and 3.x, using the latest release (0.5.18). Could you check if this could have been a temporary issue on Zenodo's side? Also, what's your version of python and dependencies (could you post the output of `pip freeze`)? 
(boutiques-dev) [glatard@sapajou ~]$ 
(boutiques-dev) [glatard@sapajou ~]$ bosh version
0.5.18

**Comment by ramou** [comment]
I have attempted using a variety of permissions on the access token and using my shell (which we'd noted sometimes causes other problems when talking to zenodo). I cannot reproduce the issue, and it appears not to be related to the access token (I figured I'd try all the variations, but it gives proper 403 errors)

```
bosh publish -v --zenodo-token sOCXT7DSs612vgxLF288XncMf2YZr9C4JrOi9ct9jabdxgv0czLz1IYetqYE --sandbox gate-opendose-descriptor.json
[ INFO ] Zenodo access token saved in /home/sthiel/.boutiques
[ INFO ] Using Zenodo endpoint https://sandbox.zenodo.org
The descriptor will be published to Zenodo, this cannot be undone. Are you sure? (Y/n) y
[ INFO (401) ] Zenodo is accessible
[ INFO (200) ] Authentication to Zenodo successful
[ INFO ] Using Zenodo endpoint https://sandbox.zenodo.org
[ INFO (200) ] Search successful.
[ INFO (201) ] Deposition succeeded, id is 261917
[ INFO (201) ] Descriptor uploaded to Zenodo
[ INFO (202) ] Descriptor published to Zenodo, doi is 10.5072/zenodo.261917
10.5072/zenodo.261917
```
I tested with Python 2.7 and the prior boutiques version (0.5.17). 

The generic 400 error, given that they're using the other error codes correctly, makes me suspicious. It would generally indicate a corrupt request. `pip freeze` might give us an idea where that's at. I'm thinking maybe _requests_ or the certificate store being out of date causing an https error? Certificate out of date errors should be given as 400, but there are a bazillion other things it could be too, so take this idea with a grain of salt.

**Comment by camarasu** [comment]
Hi everyone,

Thank you for your replies and suggestions. 
At first, I also suspected a token or Zenodo issue, but I ran tests in parellel on two unix accounts sharing the same Zenodo token, but different Boutiques installations. One of them worked, the other failed. It is still reproducible on my side. 

Here's the environment where it fails:
$ pip freeze
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
atomicwrites==1.2.1
attrs==18.2.0
backports.functools-lru-cache==1.5
backports.ssl-match-hostname==3.5.0.1
boutiques==0.5.18
certifi==2018.8.24
chardet==3.0.4
click==6.7
configobj==4.7.2
cycler==0.10.0
decorator==3.4.0
diskcache==3.0.6
fail2ban==0.9.7
funcsigs==1.0.2
functools32==3.2.3.post2
gfal2-util==1.4.0
girder-client==2.4.0
idna==2.7
iniparse==0.4
iotop==0.6
ipaddr==2.1.11
ipaddress==1.0.16
IPy==0.75
javapackages==1.0.0
jsonschema==2.6.0
kitchen==1.1.1
kiwisolver==1.0.1
langtable==0.0.31
libvirt-python==4.5.0
lxml==3.2.1
matplotlib==2.0.2
mock==2.0.0
more-itertools==4.3.0
numpy==1.15.1
pathlib2==2.3.2
pbr==5.1.2
perf==0.1
pluggy==0.7.1
policycoreutils-default-encoding==0.1
py==1.6.0
py4j==0.10.7
pycurl==7.19.0
pygobject==3.22.0
pygpgme==0.3
pyliblzma==0.5.3
pyparsing==2.2.1
pyspark==2.3.1
pytest==3.7.4
python-augeas==0.5.0
python-dateutil==2.7.3
python-dmidecode==3.10.13
python-linux-procfs==0.4.9
pytoml==0.1.18
pytz==2018.5
pyudev==0.15
pyxattr==0.5.1
PyYAML==3.13
registries==0.1
requests==2.6.0
requests-toolbelt==0.8.0
scandir==1.9.0
schedutils==0.4
seobject==0.1
sepolicy==1.1
setroubleshoot==1.1
simplejson==3.16.0
six==1.11.0
slip==0.4.0
slip.dbus==0.4.0
subprocess32==3.5.2
tabulate==0.8.3
termcolor==1.1.0
urlgrabber==3.10
urllib3==1.10.2
yum-langpacks==0.4.2
yum-metadata-parser==1.1.4

And, FYI, the one that works fine : 
$ pip freeze
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Ple                                                         ase upgrade your Python as Python 2.7 won't be maintained after that date. A fut                                                         ure version of pip will drop support for Python 2.7.
appdirs==1.4.3
asn1crypto==0.24.0
atomicwrites==1.1.5
attrs==18.1.0
backports.functools-lru-cache==1.5
backports.ssl-match-hostname==3.5.0.1
boto==2.48.0
boutiques==0.5.13
certifi==2018.4.16
cffi==1.11.5
chardet==3.0.4
click==6.7
configobj==4.7.2
configparser==3.5.0
cryptography==2.2.2
cycler==0.10.0
datalad==0.10.1
decorator==3.4.0
diskcache==3.0.6
entrypoints==0.2.3
enum34==1.1.6
fail2ban==0.9.7
fasteners==0.14.1
funcsigs==1.0.2
functools32==3.2.3.post2
gfal2-util==1.4.0
girder-client==2.4.0
gitdb2==2.0.3
GitPython==2.1.10
humanize==0.5.1
idna==2.7
iniparse==0.4
iotop==0.6
ipaddr==2.1.11
ipaddress==1.0.16
IPy==0.75
iso8601==0.1.12
javapackages==1.0.0
jsmin==2.2.2
jsonschema==2.6.0
keyring==13.1.0
keyrings.alt==3.1
kitchen==1.1.1
kiwisolver==1.0.1
langtable==0.0.31
libvirt-python==4.5.0
lxml==3.2.1
matplotlib==2.0.2
mock==2.0.0
monotonic==1.5
more-itertools==4.2.0
msgpack==0.5.6
numpy==1.15.1
pathlib2==2.3.2
patool==1.12
pbr==4.0.4
perf==0.1
pluggy==0.6.0
policycoreutils-default-encoding==0.1
py==1.5.4
py4j==0.10.7
pycparser==2.18
pycurl==7.19.0
PyGithub==1.40
pygobject==3.22.0
pygpgme==0.3
PyJWT==1.6.4
pyliblzma==0.5.3
pyparsing==2.2.1
pyspark==2.3.1
pytest==3.6.3
python-augeas==0.5.0
python-dateutil==2.7.3
python-dmidecode==3.10.13
python-linux-procfs==0.4.9
pytoml==0.1.18
pytz==2018.5
pyudev==0.15
pyxattr==0.5.1
PyYAML==3.13
registries==0.1
requests==2.19.1
requests-toolbelt==0.8.0
scandir==1.9.0
schedutils==0.4
SecretStorage==2.3.1
seobject==0.1
sepolicy==1.1
setroubleshoot==1.1
simplejson==3.15.0
six==1.11.0
slip==0.4.0
slip.dbus==0.4.0
smmap2==2.0.3
subprocess32==3.5.2
tabulate==0.8.3
termcolor==1.1.0
tqdm==4.23.4
urlgrabber==3.10
urllib3==1.23
Whoosh==2.7.4
wrapt==1.10.11
yum-langpacks==0.4.2
yum-metadata-parser==1.1.4

**Comment by ramou** [comment]
Thanks for the additional info! I've gone from slightly suspicious about requests to very suspicious about it. 

The system where things don't work is running requests 2.6.0, from March 14th, 2015. I don't know anything about what happened to that tool between then and mid-2017 (requests 2.18.4, the stated minimum requirement from requirements.txt), but if I were to attempt to debug this I'd try to force an install with the old version of requests and see if the error could be duplicated.

So I did this:
```
virtualenv i414
cd i414
source bin/activate
pip install boutiques
pip uninstall requests
pip install requests==2.6.0
bosh publish -v --sandbox gate-opendose-descriptor.json
```

And got this result:

```
[ INFO ] Zenodo access token saved in /home/sthiel/.boutiques
[ INFO ] Using Zenodo endpoint https://sandbox.zenodo.org
The descriptor will be published to Zenodo, this cannot be undone. Are you sure? (Y/n) y
[ INFO (401) ] Zenodo is accessible
[ INFO (200) ] Authentication to Zenodo successful
[ INFO ] Using Zenodo endpoint https://sandbox.zenodo.org
[ INFO (200) ] Search successful.
[ INFO (201) ] Deposition succeeded, id is 261938
[ INFO (201) ] Descriptor uploaded to Zenodo
[ ERROR (400) ] Cannot publish descriptor
```

Now, I'm not getting that nice info message before the error 400, but that still looks like a smoking gun. I assume there's a reason for the old requests, but I'd love to see what happens when you virtualenv your environment and just upgrade requests. If there's no pressing need to keep the older requests (the Internet offered one example of doing it to make something else keep working on an old mac, but that was many years ago), then I'd suggest updating it.

**Comment by axlbonnet** [comment]
> Now, I'm not getting that nice info message before the error 400, but that still looks like a smoking gun. I assume there's a reason for the old requests, but I'd love to see what happens when you virtualenv your environment and just upgrade requests. If there's no pressing need to keep the older requests (the Internet offered one example of doing it to make something else keep working on an old mac, but that was many years ago), then I'd suggest updating it.

I added the nice info message myself by editing the source code, from @erinb90 's suggestion to have more information about the issue.

I updated requests and now it's working, thanks !

**Comment by ramou** [comment]
Ah yes, I recall the exchange.

Very glad to have helped!

**Comment by glatard** [comment]
This should be solved in 0.5.19 as we now have `requests==2.18.4`.
